### PR TITLE
[DataTableHeaderCell] Hover Tooltip in new design, [ColumnConfigurati…

### PR DESCRIPTION
**Pull request by vik753**
…onModal] added renderItem callback to render label

### CODE REVIEW

File: uui/components/tables/DataTableHeaderCell.tsx [hunk 1]
```diff
@@ -30,28 +25,29 @@ export class DataTableHeaderCell<TItem, TId> extends React.Component<DataTableHe
...
-            <Tooltip
-                trigger="hover"
-                placement="bottom-start"
-                renderContent={ !this.state.isDropdownOpen && this.props.column.info ? () => this.props.column.info : null }
-                color="default"
-                cx={ css.cellTooltip }
-                offset={ [-12, 12] }
-            >
+    getColumnCaption = () => {
+        const getTooltipContent = () => !this.state.isDropdownOpen && (
```
Comment (AlekseyManetov) [inline uui/components/tables/DataTableHeaderCell.tsx:29]: Let's move the declaration outside getColumnCaption function

File: uui/components/tables/columnsConfigurationModal/ColumnRow.tsx [hunk 2]
```diff
@@ -33,11 +34,31 @@ export const ColumnRow = React.memo(function ColumnRow(props: ColumnRowProps) {
             ...(isDndAllowed ? { onTouchStart, onPointerDown } : {}),
         };
 
+        const getLabelContent = (props: ColumnsConfigurationRowProps) => {
+            return (
+                <FlexRow>
+                    <span>{props.caption}</span>
+                    { props.info && (
+                        <span style={ { fontSize: '10px', marginTop: '2px', marginLeft: '3px', color: 'gray' } }>
```
Comment (AlekseyManetov) [inline uui/components/tables/columnsConfigurationModal/ColumnRow.tsx:42]: Why we use hire inline styles? Let's move it to the css class.

File: uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx [hunk 3]
```diff
@@ -73,7 +73,21 @@ export function ColumnsConfigurationModal<TItem, TId, TFilter>(props: ColumnsCon
                 {!!amountPinned && (
                     <FlexRow cx={ styles.groupItems } size="30">
                         {groupedColumns.displayedPinned.map((c) => (
-                            <ColumnRow column={ c } key={ c.key } />
+                            <ColumnRow
+                                renderItem={ () => (
+                                    <FlexRow>
```
Comment (AlekseyManetov) [inline uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx:78]: Why we need it here?  Is it temporary code for debug?

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui/components/tables/DataTableHeaderCell.tsx (hunk 1 above)
Comment (vik753) [reply uui/components/tables/DataTableHeaderCell.tsx:29]: done

File: uui/components/tables/columnsConfigurationModal/ColumnRow.tsx (hunk 2 above)
Comment (vik753) [reply uui/components/tables/columnsConfigurationModal/ColumnRow.tsx:42]: done

File: uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx (hunk 3 above)
Comment (vik753) [reply uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx:78]: Yes, it was for debugging.

File: uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts [hunk 4]
```diff
@@ -49,23 +49,30 @@ function isSubstring(s: React.ReactNode, sub: string) {
     }
     return false;
 }
-export function isColumnFilteredOut(c: DataColumnProps, filter?: string) {
-    const caption = c.caption;
+
+export function isColumnFilteredOut(column: any, searchFields: string[], filter?: string) {
```
Comment (AlekseyManetov) [inline uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts:53]: avoid any here

File: uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts [hunk 5]
```diff
@@ -49,23 +49,30 @@ function isSubstring(s: React.ReactNode, sub: string) {
...
     const accUnsorted = {
         displayedPinned: [],
         displayedUnpinned: [],
         hidden: [],
     } as GroupedColumnsType;
 
-    return sortedColumns.reduce((acc, i) => {
-        if (!isColumnFilteredOut(i, searchValue)) {
+    return props.sortedColumns.reduce((acc, i) => {
+        if (!isColumnFilteredOut(i, props?.getSearchFields ? props.getSearchFields(i) : [i.caption as string], props.searchValue)) {
```
Comment (AlekseyManetov) [inline uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts:75]: lets move this default inside UseColumnsConfiguration

File: uui/components/tables/DataTable.tsx [hunk 6]
```diff
@@ -72,21 +54,27 @@ export function DataTable<TItem, TId>(props: React.PropsWithChildren<DataTablePr
     }, [props.renderNoResultsBlock]);
 
     const onConfigurationButtonClick = React.useCallback(() => {
+        const outerProps = { columns: props.columns, columnsConfig: { ...config }, defaultConfig };
```
Comment (AlekseyManetov) [inline uui/components/tables/DataTable.tsx:57]: rename just to 'modalProps' and pass it also in our default implementation

File: uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx [hunk 7]
```diff
@@ -73,15 +57,15 @@ export function ColumnsConfigurationModal<TItem, TId, TFilter>(props: ColumnsCon
                 {!!amountPinned && (
                     <FlexRow cx={ styles.groupItems } size="30">
                         {groupedColumns.displayedPinned.map((c) => (
-                            <ColumnRow column={ c } key={ c.key } />
+                            <ColumnRow column={ c } key={ c.key } renderItem={ props.renderItem ? () => props.renderItem(c) : null } />
```
Comment (AlekseyManetov) [inline uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx:60]: why do we have `props.renderItem ? () => props.renderItem(c) : null` condition here, if we have the same inside `ColumnRow`?

File: uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx [hunk 8]
```diff
@@ -73,15 +57,15 @@ export function ColumnsConfigurationModal<TItem, TId, TFilter>(props: ColumnsCon
...
                         ))}
                     </FlexRow>
                 )}
                 {hasDivider && <div className={ styles.hDivider } />}
                 {!!amountUnPinned && (
                     <FlexRow cx={ styles.groupItems } size="30">
                         {groupedColumns.displayedUnpinned.map((c) => (
-                            <ColumnRow column={ c } key={ c.key } />
+                            <ColumnRow column={ c } key={ c.key } renderItem={ props.renderItem ? () => props.renderItem(c) : null } />
```
Comment (AlekseyManetov) [inline uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx:68]: why do we have `props.renderItem ? () => props.renderItem(c) : null` condition here, if we have the same inside `ColumnRow`?

File: uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx [hunk 9]
```diff
@@ -98,7 +82,7 @@ export function ColumnsConfigurationModal<TItem, TId, TFilter>(props: ColumnsCon
                 {renderGroupTitle(i18n.hiddenSectionTitle, amountHidden)}
                 <FlexRow cx={ styles.groupItems } size="30">
                     {groupedColumns.hidden.map((c) => (
-                        <ColumnRow column={ c } key={ c.key } />
+                        <ColumnRow column={ c } key={ c.key } renderItem={ props.renderItem ? () => props.renderItem(c) : null } />
```
Comment (AlekseyManetov) [inline uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx:85]: why do we have `props.renderItem ? () => props.renderItem(c) : null` condition here, if we have the same inside `ColumnRow`?

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts (hunk 4 above)
Comment (vik753) [reply uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts:53]: fixed

File: uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts (hunk 5 above)
Comment (vik753) [reply uui-components/src/table/columnsConfigurationModal/columnsConfigurationUtils.ts:75]: done

File: uui/components/tables/DataTable.tsx (hunk 6 above)
Comment (vik753) [reply uui/components/tables/DataTable.tsx:57]: we already have props and modalProps here, so I renamed it into configProps.

File: uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx (hunk 8 above)
Comment (vik753) [reply uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx:68]: fixed it

File: uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx (hunk 9 above)
Comment (vik753) [reply uui/components/tables/columnsConfigurationModal/ColumnsConfigurationModal.tsx:85]: fixed it

**Comment by vik753** [comment]
https://github.com/epam/UUI/issues/1423

**Pull request merged**